### ISSUE TITLE
Models will be used

### ISSUE THREAD

**Issue by yalsaffar** [Google-Solution-2023/Air-Pollution-Detector#3]
I am writing this issue to document the approach I will take in constructing the predictive models.
The core idea of the predictive models will be focused on training different models with different architectures, which are the following:

- A simple regression model that does not take time-series attributes into consideration. My research has shown that only knowing some variables of the weather and the station; which can predict with good precision(Catboost or XGBoost).
- A window regression where it will be separated by element/station where it will take a window of previous records of every element and for every station to predict the future values. This also has been shown effective to some extent.
- LSTM will be tested and fine-tuned to try to use this NN time-series approach which is proven to be very effective for our cause, and especially for problems such as pollution prediction. However, it could be challenging with the limited data we have at this stage. More research and advanced testing needed to be done before deciding.
- The prophet model will be essential in the architecture, The main advantage of using this model is that it is very well suited for seasonal data, and actually so research on how it is useful for air pollution prediction. My initial tests have shown that it is very fast and quite accurate.

The predictions of all these models will be combined, mainly soft ensemble models(still need to test different approaches here).
Let me know if you have any suggestions or questions.
